sql/src/main/java/io/crate/operation/reference/sys/check/cluster/TablesNeedUpgradeSysCheck.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.operation.reference.sys.check.cluster;
+
+import io.crate.action.sql.Option;
+import io.crate.action.sql.ResultReceiver;
+import io.crate.action.sql.SQLOperations;
+import io.crate.data.Row;
+import io.crate.metadata.Schemas;
+import io.crate.operation.reference.sys.check.AbstractSysCheck;
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.cluster.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.inject.Singleton;
+import org.elasticsearch.common.logging.ESLogger;
+import org.elasticsearch.common.logging.Loggers;
+
+import javax.annotation.Nonnull;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+@Singleton
+public class TablesNeedUpgradeSysCheck extends AbstractSysCheck {
+
+    public static final int ID = 4;
+    public static final String DESCRIPTION =
+        "The following tables must be upgraded for compatibility with future versions of CrateDB: ";
+
+    private static final ESLogger LOGGER = Loggers.getLogger(TablesNeedUpgradeSysCheck.class);
+    private static final String STMT = "select schema_name || '.' || table_name, min_lucene_version " +
+                                       "from sys.shards where min_lucene_version <> '" +
+                                       org.apache.lucene.util.Version.LATEST.toString() + "'";
+    private static final String STMT_NOT_IN = " AND schema_name || '.' || table_name NOT IN (";
+    private static final String STMT_SUFFIX = " order by 1";
+    private static final int LIMIT = 50_000;
+
+    private final SQLOperations.Session session;
+    private final ClusterService clusterService;
+    private final Schemas schemas;
+    private volatile Collection<String> tablesNeedUpgrade;
+
+    @Inject
+    public TablesNeedUpgradeSysCheck(ClusterService clusterService, Schemas schemas, SQLOperations sqlOperations) {
```

Comment:
What's the reason that this wasn't implemented as node check so that it doesn't have to do requests?